crates/miden-objects/src/account/builder/mod.rs

```diff
@@ -100,15 +116,21 @@ impl AccountBuilder {
         #[cfg(all(not(feature = "testing"), not(test)))]
         let vault = AssetVault::default();
 
-        let (code, storage) =
-            Account::initialize_from_components(self.account_type, &self.components).map_err(
-                |err| {
-                    AccountError::BuildError(
-                        "account components failed to build".into(),
-                        Some(Box::new(err)),
-                    )
-                },
-            )?;
+        let auth_component = self
+            .auth_component
+            .take()
+            .ok_or(AccountError::BuildError("auth component must be set".into(), None))?;
+
+        let mut components = vec![auth_component];
+        components.extend(core::mem::take(&mut self.components));
```

Comment:
nit: it may be possible to write this as:
```rust
components.append(&mut self.components);
```

Reply:
Ah yes, this is possible now that `build_inner` takes a mutable reference to `self`. Applied in [append instead of extend](https://github.com/0xMiden/miden-base/pull/1480/commits/920c4ecb6b1ca7b8285acb416564025b92b43995)